GUI/Forms/ExtractProgressForm.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using SteamDatabase.ValvePak;
+
+namespace GUI.Forms
+{
+    public partial class ExtractProgressForm : Form
+    {
+        private CancellationTokenSource cancellationTokenSource;
+        private Package package;
+        private TreeNode root;
+        private string path;
+        private Queue<PackageEntry> filesToExtract;
+        private int initialFileCount;
+
+        public ExtractProgressForm(Package package, TreeNode root, string path)
+        {
+            InitializeComponent();
+
+            cancellationTokenSource = new CancellationTokenSource();
+            filesToExtract = new Queue<PackageEntry>();
+            initialFileCount = 0;
+
+            this.package = package;
+            this.root = root;
+            this.path = path;
+        }
+
+        protected override void OnShown(EventArgs e)
+        {
+            Task
+                .Run(async () =>
+                {
+                    Invoke((Action)(() =>
+                    {
+                        extractStatusLabel.Text = "Calculating...";
+                        extractProgressBar.Style = ProgressBarStyle.Marquee;
+                    }));
+
+                    await CalculateFilesToExtractAsync(root);
+                    initialFileCount = filesToExtract.Count;
+
+                    Invoke((Action)(() =>
+                    {
+                        extractProgressBar.Style = ProgressBarStyle.Continuous;
+                    }));
+
+                    await ExtractFilesAsync();
+                });
+        }
+
+        protected override void OnClosing(CancelEventArgs e)
+        {
+            cancellationTokenSource.Cancel();
+        }
+
+        private async Task CalculateFilesToExtractAsync(TreeNode root)
```

Comment:
This is async but doesn't actually do any async work?

Reply:
Sorry, the habit of always making everything async :)